coralproject/talk
# Fix popup blocked issue

## What does this PR do?

The main change is that this updates the `Popup` component to use `useLayoutEffect` instead of `useEffect` for opening the popup window if it should open and hasn't already been opened (I added a comment in the file about this as well). The reason that we need to use this is so that this happens synchronously and the popup opening is tied to the user's click event clicking the sign in button, as otherwise, if we use `useEffect` and it happens asynchronously, the event is not tied to the user and browsers (Firefox) will try to block the popup and prompt the user to confirm it instead of just opening it.

I also did some more refactoring while here. Updated to use `useRef` to keep track of the ref for the popup window, updated to destructure the props for the `Popup` component for readability, and updated with more `useCallback`.

## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags? 

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

In Firefox, go to the stream (ensure that in Firefox popup blocker settings, your localhost is not included in its Exceptions). Sign out if you are signed in. Click `Sign in`. See that the popup opens without any prompt/blocking attempt. Test out the sign in in general to make sure everything related to it is functioning as expected.
 
## How do we deploy this PR?
